test/TestProjects/FirstTest-TypeSpec/src/Generated/FirstTestTypeSpecClient.cs

```diff
@@ -678,6 +678,76 @@ public virtual Response StringFormat(Guid subscriptionId, RequestContent content
             }
         }
 
+        /// <summary>
+        /// [Protocol Method] head as boolean.
+        /// <list type="bullet">
+        /// <item>
+        /// <description>
+        /// This <see href="https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/core/Azure.Core/samples/ProtocolMethods.md">protocol method</see> allows explicit creation of the request and processing of the response for advanced scenarios.
+        /// </description>
+        /// </item>
+        /// </list>
+        /// </summary>
+        /// <param name="id"> The String to use. </param>
+        /// <param name="context"> The request context, which can override default behaviors of the client pipeline on a per-call basis. </param>
+        /// <exception cref="ArgumentNullException"> <paramref name="id"/> is null. </exception>
+        /// <exception cref="ArgumentException"> <paramref name="id"/> is an empty string, and was expected to be non-empty. </exception>
+        /// <exception cref="RequestFailedException"> Service returned a non-success status code. </exception>
+        /// <returns> The response returned from the service. </returns>
+        /// <include file="Docs/FirstTestTypeSpecClient.xml" path="doc/members/member[@name='HeadAsBooleanAsync(string,RequestContext)']/*" />
+        public virtual async Task<Response<bool>> HeadAsBooleanAsync(string id, RequestContext context = null)
```

Comment:
Is this design reviewed by anyone?
Currently, in our protocol methods, no matter protocol method returns or does not return a body, we always use `Response` as return type, we never have `Response<BinaryData>`.
In this case, it is similar - if we follow the pattern, we should return Response in protocols and return `Response<bool>` in convenience maybe?

Reply:
@ArcturusZhang It is consistent with previous design. see `HeadAsBooleanTrue-LowLevel`. We don't change it.